compute_endpoint/tests/unit/test_config_utils.py

```diff
@@ -435,6 +437,25 @@ def test_render_config_passes_mapped_identity(mocker, conf_no_exec):
     assert rendered_dict["globus_id"] == mock_matched_identity
 
 
+def test_render_config_passes_all_reserved_variables(
+    conf_no_exec, mapped_ident: MappedPosixIdentity
+):
+    template = ""
+    for var in RESERVED_USER_CONFIG_TEMPLATE_VARIABLES:
+        template += f"""
+{{% if {var} is defined %}}
+# "{var}" is defined
+{{% endif %}}
+"""
```

Comment:
- From a _unit_ test perspective, this is a good opportunity to parameterize the test function:

    ```python
    @pytest.mark.parametrize("res_var", RESERVED_USER_CONFIG_TEMPLATE_VARIABLES)
    def test_render_config_passes...variables(res_var, ...)
    ```
- Strongly consider a parallel null test.  I often call these "sanity checks," depending on context.  Here, I'd verify that the templating is doing what it's supposed to and _not_ including the variable if it's not defined.  This could be as simple as a known-to-the-test-does-not-exist-variable.

Reply:
test is now parametrized 👍 

> not including the variable if it's not defined

I'll address that in a follow-up PR - briefly, it's only possible for `user_runtime` to be null, but we currently paper over that and include an empty dict instead of not setting it

Reply:
following an out of band conversation, also just added a "make sure something weird isn't rendered" step to the test